### PR TITLE
[#38] create association between two models and 3rd model

### PR DESCRIPTION
**Pull request by abukowska**
@sandrakoryszewska, please check it. Thank you.

### CODE REVIEW

File: app/models/ingredient.rb [hunk 1]
```diff
@@ -3,4 +3,7 @@ class Ingredient < ApplicationRecord
     uniqueness: true,
     format: { with: /\A[a-z][a-z\s]+[a-z]\z/,
     message: "only allows lowercase letters and whitespace" }
+
+  has_many :recipe_and_ingredients
+  has_many :recipes, through: :recipe_and_ingredients
```
Comment (CallMeSandra) [inline app/models/ingredient.rb:8]: :+1:

File: app/models/recipe.rb [hunk 2]
```diff
@@ -2,4 +2,7 @@ class Recipe < ApplicationRecord
   validates :name, presence: true,
     uniqueness: true
   validates :description, presence: true
+
+  has_many :recipe_and_ingredients
+  has_many :ingredients, through: :recipe_and_ingredients
 end
```
Comment (CallMeSandra) [inline app/models/recipe.rb:8]: :+1:

File: app/models/recipe_and_ingredient.rb [hunk 3]
```diff
@@ -0,0 +1,4 @@
+class RecipeAndIngredient < ApplicationRecord
+  belongs_to :recipe
+  belongs_to :ingredient
+end
```
Comment (CallMeSandra) [inline app/models/recipe_and_ingredient.rb:4]: :+1:

File: db/migrate/20160721111955_create_recipe_and_ingredients.rb [hunk 4]
```diff
@@ -0,0 +1,6 @@
+class CreateRecipeAndIngredients < ActiveRecord::Migration[5.0]
+  def change
+    add_column :recipe_and_ingredients, :ingredient_id, :integer
+    add_column :recipe_and_ingredients, :recipe_id, :integer
+  end
+end
```
Comment (CallMeSandra) [inline db/migrate/20160721111955_create_recipe_and_ingredients.rb]: :+1:

File: Gemfile.lock [hunk 5]
```diff
@@ -89,7 +89,7 @@ GEM
       pkg-config (~> 1.1.7)
     pg (0.18.4)
     pkg-config (1.1.7)
-    puma (3.4.0)
+    puma (3.5.0)
     rack (2.0.1)
     rack-test (0.6.3)
       rack (>= 1.0)
```
Comment (CallMeSandra) [inline Gemfile.lock:95]: :+1:

**Comment by agatasalska** [comment]
there are two files with migration for this model: 20160721111955_create_recipe_and_ingredients.rb
and 20160720112100_create_recipe_and_ingredients.rb. Which one is valid? The earlier one looks better. Do rake db:rollback or rake db:reset, choose the correct file and run rake db:migrate.

**Comment by CallMeSandra** [comment]
Delete every file that doesn't have :+1:

**Comment by agatasalska** [comment]
should be ok now 👍

**Comment by agatasalska** [comment]
@gotar check pls

File: config/routes.rb [hunk 6]
```diff
@@ -1,4 +1,5 @@
 Rails.application.routes.draw do
+  resources :recipe_and_ingredients
```
Comment (gotar) [inline config/routes.rb]: Hmm you don't need any new route. You can drop it

File: db/migrate/20160721111955_create_recipe_and_ingredients.rb (hunk 4 above)
Comment (gotar) [inline db/migrate/20160721111955_create_recipe_and_ingredients.rb]: It's ok but better will be to add relations type not integer

File: spec/models/recipe_and_ingredient_spec.rb [hunk 7]
```diff
@@ -0,0 +1,5 @@
+require 'rails_helper'
+
+RSpec.describe RecipeAndIngredient, type: :model do
+  pending "add some examples to (or delete) #{__FILE__}"
+end
```
Comment (gotar) [inline spec/models/recipe_and_ingredient_spec.rb]: Drop empty spec

**Comment by gotar** [comment]
You need to squash after you fix and rebase with master

**Pull request merged**